R/map.R

```diff
@@ -169,13 +169,46 @@ map_ <- function(.type,
   names <- vec_names(.x)
 
   .f <- as_mapper(.f, ...)
-
   i <- 0L
+  print(i)
+
+  the$last_map_index <- NULL
+  the$last_map_results <- NULL
+  the$last_map <- list(
+    env = current_env(),
+    call = expr(call_with_cleanup(map_impl, environment(), .type, .progress, n, names, i, the))
+  )
+
   with_indexed_errors(
     i = i,
     names = names,
     error_call = .purrr_error_call,
-    call_with_cleanup(map_impl, environment(), .type, .progress, n, names, i)
+    call_with_cleanup(map_impl, environment(), .type, .progress, n, names, i, the)
+  )
+}
+
+purrr_continue <- function(.f = NULL) {
```

Comment:
It would be even nicer if updating the original `.f` would work, i.e.

```r
map(x, f)
# something went wrong

# update `f`
f <- function(x) ...
# no need to pass the new `f` to `purrr_continue()`
purrr_continue()
```